espaloma/data/dataset.py

```diff
@@ -0,0 +1,186 @@
+# =============================================================================
+# IMPORTS
+# =============================================================================
+import espaloma as esp
+import abc
+import torch
+
+# =============================================================================
+# MODULE CLASSES
+# =============================================================================
+class Dataset(abc.ABC, torch.utils.data.Dataset):
+    """ The base class of map-style dataset.
+
+    Parameters
+    ----------
+    graphs : list
+        objects in the dataset
+
+    Note
+    ----
+    This also supports iterative-style dataset by deleting `__getitem__`
+    and `__len__` function. 
+
+    Attributes
+    ----------
+    transforms : an iterable of callables that transforms the input.
```

Comment:
Add a bit more description or example here -- for example, the `__getitem__` definition applies these transforms sequentially, so order may matter